src/components/SideNav.jsx

```diff
@@ -1,20 +1,48 @@
 import { NavLink } from "react-router-dom";
+import { useDBdata } from "../context/db-data-context";
 import { useMessageHandling } from "../context/message-handling-context";
 import "../stylesheets/sidenav.css";
 import { sidenavItemList } from "../utilities/sidenav-items";
+import { v4 as uuid } from "uuid";
+
 export function SideNav() {
+  const { labels } = useDBdata();
   const { showSidenav, setShowSidenav } = useMessageHandling();
   return (
     <aside
       className={`sidenav-bg ${!showSidenav ? "sidenav-hide" : ""}`}
       onClick={() => setShowSidenav(false)}
     >
       <ul className="list-group-stacked sidenav">
-        {sidenavItemList.map((item) => (
+        <NavLink
+          end={true}
+          to="/notes"
+          key={uuid()}
+          className={({ isActive }) => (isActive ? "link active" : "link")}
+        >
+          <li className="list-item list-w-icon px-2">
+            <i className="material-icons-outlined list-icon">description</i>
+            Notes
+          </li>
+        </NavLink>
+        {labels?.map((item) => (
+          <NavLink
+            end={true}
+            to={`/notes/label/${item}`}
+            key={uuid()}
+            className={({ isActive }) => (isActive ? "link active" : "link")}
```

Comment:
why is callback used in className and from where are you getting isActive as parameters?

Reply:
isActive state is returned by Navlink itself, which can be used for breadcrumbs especially to style navigation based on whether a user is on a particular page

Reply:
That's how it is supposed to be for NavLink
Reference: https://reactrouter.com/docs/en/v6/api#navlink